pkg/storegateway/series_chunks.go

```diff
@@ -320,9 +364,50 @@ func (c *loadingSeriesChunksSetIterator) Next() (retHasNext bool) {
 	// This data structure doesn't retain the seriesChunkRefsSet so it can be released once done.
 	defer nextUnloaded.release()
 
+	c.chunkReaders.reset()
+	groupCount := 0
+	for _, s := range nextUnloaded.series {
+		groupCount += len(s.groups)
+	}
+	loadedGroups := make([][]byte, groupCount)
```

Comment:
keeping track of this slice along with the `var currentGroup int` counter and using `nextUnloaded.series` makes all this a bit clunky. I will add an intermediate type